app/src/main/java/org/openobservatory/ooniprobe/test/test/AbstractTest.java

```diff
@@ -87,10 +86,7 @@ void run(Context c, PreferenceManager pm, Gson gson, Settings settings, Result r
 						measurement.save();
 						break;
 					case "log":
-						if (logFOS == null)
-							logFOS = new FileOutputStream(Measurement.getLogFile(c, result.id, name));
-						logFOS.write(event.value.message.getBytes());
-						logFOS.write('\n');
+						FileUtils.writeStringToFile(Measurement.getLogFile(c, result.id, name), event.value.message + "\n", StandardCharsets.UTF_8, true);
```

Comment:
Maybe adding a comment saying that `true` means we are appending to the file is useful.

Though this may just be my lack of familiarity with these java function calls.